promise.go

```diff
@@ -200,41 +200,48 @@ func (promise *Promise) Await() (interface{}, error) {
 // from the resolved promises in the same order as defined in the iterable of multiple promises.
 // If it rejects, it is rejected with the reason from the first promise in the iterable that was rejected.
 func All(promises ...*Promise) *Promise {
-	var length = len(promises)
+	psLen := len(promises)
 
-	if promises == nil || length == 0 {
-		return Resolve(make([]interface{}, 0))
+	if psLen == 0 {
+		return Resolve([]interface{}{})
 	}
 
 	return New(func(resolve func(interface{}), reject func(error)) {
-		var resolutionsChan = make(chan []interface{}, length)
-		var errorChan = make(chan error, length)
+		resolvedChan := make(chan []interface{}, psLen)
+		rejectedChan := make(chan error, psLen)
 
-		for index, promise := range promises {
-			func(x int) {
-				promise.Then(func(data interface{}) interface{} {
-					resolutionsChan <- []interface{}{x, data}
+		for ind, promise := range promises {
+			func (i int, p *Promise) {
+				p.Then(func(data interface{}) interface{} {
+					resolvedChan <- []interface{}{i, data}
 					return data
-				})
-				promise.Catch(func(err error) error {
-					errorChan <- err
+				}).Catch(func(err error) error {
+					rejectedChan <- err
 					return err
 				})
-			}(index)
+			}(
+				ind,
+				promise,
+			)
 		}
 
-		var resolutions = make([]interface{}, length)
-		for x := 0; x < length; x++ {
+		resolutions := make([]interface{}, psLen)
+		numResolved := 0
+
+		for {
+			if numResolved == psLen {
+				resolve(resolutions)
+				return
+			}
 			select {
-			case resolution := <-resolutionsChan:
+			case resolution := <- resolvedChan:
 				resolutions[resolution[0].(int)] = resolution[1]
-
-			case err := <-errorChan:
+				numResolved += 1
+			case err := <-rejectedChan:
```

Comment:
After rebase makes it look like I just went ahead and renamed all your variables lol will rename mine to yours :-D :)

Reply:
Alright 👍